### PR TITLE
Run recommended config against source code

### PR DESCRIPTION
**Pull request by fisker**
Shouldn't we run recommended configs on our own repo? And our own source code should follow recommended rules.

This is a prototype, maybe move to integration test

### CODE REVIEW

**Comment by sindresorhus** [comment]
Makes sense to me. You need to add `configs` to `files` in package.json.

**Comment by fisker** [comment]
Just a idea, maybe a better way, will do it later

**Comment by fisker** [comment]
This PR is almost done.

Some source code not following recommended configs, most of them are fixed, some questions:

1) `prevent-abbreviations` default `checkProperties: true` is annoying, I override it for now, maybe we should consider to change that, #271 #367

2) `prevent-abbreviations` also complain about `get-docs-url.js` filename, I can rename it to fix that, and I suggest we do this to follow `recommended` config, just lots of files, should I do that?

3) some comments fails on `expiring-todo-comments` rule, I can add `eslint-disable` but `xo` will reports another error

4) `consistent-function-scoping` seems buggy?

/cc @sindresorhus

**Comment by fisker** [comment]
/cc @sindresorhus 

Please have a look at the problems. 
I need your opinion on them to continue.

**Comment by sindresorhus** [comment]
1. As commented in the linked issue, let's change the default.
2. Yes, rename it.
3. Fixed by https://github.com/sindresorhus/eslint-plugin-unicorn/pull/400
4. Disable it for now.

File: docs/new-rule.md [hunk 1]
```diff
@@ -24,5 +24,6 @@ Use the [`astexplorer` site](https://astexplorer.net) with the `espree` parser a
 	*(The description should be the same as the heading of the documentation file).*
 - Run `$ npm test` to ensure the tests pass.
 - Run `$ npm run integration` to run the rules against real projects to ensure your rule does not fail on real-world code.
+- Run `$ npm run lint` to run the rules against this repository to ensure code in the repository are following your rule.
```
Comment (sindresorhus) [inline docs/new-rule.md]: Why a separate step? It should just be part of `npm test`.

**Comment by sindresorhus** [comment]
Can you fix the merge conflict

File: docs/new-rule.md (hunk 1 above)
Comment (fisker) [reply docs/new-rule.md]: I was thinking, when some one making a new rule, they can do this after their PR is reviewed, new rules might requires other code changes

Comment (sindresorhus) [reply docs/new-rule.md]: Good point. Ok.

**Comment by fisker** [comment]
`lint.js` can't pass xo, because I'm not using ava?

File: rules/catch-error-name.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
 'use strict';
 const astUtils = require('eslint-ast-utils');
 const avoidCapture = require('./utils/avoid-capture');
-const getDocsUrl = require('./utils/get-docs-url');
+const getDocumentsUrl = require('./utils/get-documents-url');
```
Comment (futpib) [inline rules/catch-error-name.js]: I think `docs` here stands for `documentation` rather than for `documents`.

Comment (fisker) [reply rules/catch-error-name.js]: So, I should rename them again....

Comment (futpib) [reply rules/catch-error-name.js]: I recommend using [`rg`](https://github.com/BurntSushi/ripgrep) and `perl` for batch replacements.

```bash
rg 'getDocumentsUrl' -l | xargs perl -i -pe 's/getDocumentsUrl/getDocumentationUrl/g'
rg 'get-documents-url' -l | xargs perl -i -pe 's/get-documents-url/get-documentation-url/g'
```

Comment (fisker) [reply rules/catch-error-name.js]: done.

**Comment by fisker** [comment]
@futpib any idea why CI fails?

File: rules/catch-error-name.js (hunk 2 above)
Comment (fisker) [reply rules/catch-error-name.js]: @futpib I did this with vscode `replace in file`(not sure what this called, using other lang), after I saw you comment, remember me that sindresorhus has a new cli [`replace-in-files-cli`](https://github.com/sindresorhus/replace-in-files-cli)

**Comment by fisker** [comment]
How the same code success on my CI? 

https://travis-ci.com/fisker/eslint-plugin-unicorn/builds/131622273

File: .travis.yml [hunk 3]
```diff
@@ -3,11 +3,17 @@ node_js:
   - '12'
   - '10'
   - '8'
+cache:
+  npm: false
```
Comment (fisker) [inline .travis.yml]: CI finally passed, should I keep this?

Comment (sindresorhus) [reply .travis.yml]: Sure, for now.

**Pull request merged**